### ISSUE TITLE
ARRAY datatype not being parsed correctly with Trino dialect

### ISSUE THREAD

**Issue by kirkhansen** [sqlfluff/sqlfluff#5721]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.


### What Happened

I had some SQL making use of the ARRAY type and got some unexpected results. Boiling it down to a simple example:

```sql
SELECT ARRAY[ARRAY[1, 2], ARRAY[3, 4]];
```

returns

```bash                                                                                                                                                                                                                                                      sqlfluff lint --dialect=trino ../query.sql
== [../query.sql] FAIL                                                                                                                                                                                                                                                                    
L:   1 | P:  21 |  PRS | Line 1, Position 21: Found unparsable section: ', 2'
L:   1 | P:  25 |  PRS | Line 1, Position 25: Found unparsable section: ',
                       | ARRAY[3, 4]'
WARNING: Parsing errors found and dialect is set to 'trino'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### Expected Behaviour

SQLFluff should recognize ARRAY as a type and not error out.

### Observed Behaviour

SQLFLuff errors out unable to parse a query that works with Trino.

### How to reproduce

Create a one-line SQL file called query.sql with the following contents:

```sql
SELECT ARRAY[ARRAY[1, 2], ARRAY[3, 4]];
```

Then run the following command with the latest SQLFluff release:

```bash
sqlfluff lint --dialect=trino ../query.sql
```

### Dialect

trino

### Version

```bash
sqlfluff --version
sqlfluff, version 2.3.5
```

```
python --version
Python 3.11.8
```

### Configuration

No configuration file was used. This can be triggered with the default configuration.

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
